meshmode/array_context.py

```diff
@@ -213,9 +229,11 @@ def __init__(self):
     def _get_fake_numpy_namespace(self):
         return _BaseFakeNumpyNamespace(self)
 
+    @abstractmethod
     def empty(self, shape, dtype):
         raise NotImplementedError
```

Comment:
```suggestion
        pass
```
? I think abstract methods have their own exception when not implemented?

Reply:
Ah, you just can't instantiate them at all, so I guess this makes no difference.